pkgs/development/python-modules/mypy/default.nix

```diff
@@ -1,28 +1,62 @@
-{ lib, stdenv, fetchPypi, buildPythonPackage, typed-ast, psutil, isPy3k
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, buildPythonPackage
 , mypy-extensions
+, psutil
+, python
+, pythonOlder
+, typed-ast
 , typing-extensions
+, toml
+, tomli
+, types-toml
+, types-typed-ast
 }:
+
 buildPythonPackage rec {
   pname = "mypy";
-  version = "0.812";
-  disabled = !isPy3k;
+  version = "2021-09-17";
+  disabled = pythonOlder "3.6";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "069i9qnfanp7dn8df1vspnqb0flvsszzn22v00vj08nzlnd061yd";
+  src = fetchFromGitHub {
+    owner = "python";
+    repo = "mypy";
+    rev = "c6e8a0b74c32b99a863153dae55c3ba403a148b5";
+    sha256 = "sha256-/hlGkz3o7S8fBrvl0sl4J3LBpbVs1QhiRj9irKgajOs=";
   };
 
-  propagatedBuildInputs = [ typed-ast psutil mypy-extensions typing-extensions ];
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/python/mypy/commit/f1755259d54330cd087cae763cd5bbbff26e3e8a.patch";
+      sha256 = "1gwxjxchn22dk7iqfykghgpxh6mqwm9fng9pkbq5bayyfp96s3b5";
+    })
+  ];
+
+  nativeBuildInputs = [
+    types-toml
+    types-typed-ast
+  ];
+
+  propagatedBuildInputs = [
+    mypy-extensions
+    psutil
+    toml
+    tomli
```

Comment:
You are sure we need both? Maybe a comment could clarify why this is needed.

Reply:
It errored without it. Try removing it and you'll see. Unfortunately I'm not a Python programmer so I'm not sure of most things...